vault/core.go

```diff
@@ -2632,6 +2647,91 @@ func (c *Core) SetConfig(conf *server.Config) {
 	c.logger.Debug("set config", "sanitized config", string(bz))
 }
 
+func (c *Core) getCustomHeadersListenerList(listenerAdd string) []*ListenerCustomHeaders {
+	customHeaders := c.customListenerHeader.Load()
+	if customHeaders == nil {
+		return nil
+	}
+
+	customHeadersList := customHeaders.([]*ListenerCustomHeaders)
+	if customHeadersList == nil {
+		return nil
+	}
+
+	// either looking for a specific listener, or if listener address isn't given,
+	// checking for all available listeners
+	var lch []*ListenerCustomHeaders
+	if listenerAdd == "" {
+		lch = customHeadersList
+	} else {
+		for _, l := range customHeadersList {
+			if l.Address == listenerAdd {
+				lch = append(lch, l)
+				break
+			}
+		}
+		if len(lch) == 0 {
+			return nil
+		}
+	}
+	return lch
+}
+
+func (c *Core) GetListenerCustomResponseHeaders(listenerAdd string) *ListenerCustomHeaders {
```

Comment:
Not sure how to add to an outdated conversation, so re:

> The "if address is an empty string" comment is in another function though, not sure what to do about that here? would you please elaborate?

I just think it's worth noting in godoc on this method that listenerAddr is optional.

Reply:
I see. I will add it to the ExistCustomResponseHeader function